In [7]:
import math
import scipy.stats as st
import numpy as np

# Práctica 4

## 1. El propietario de Britten’s Egg Farm desea calcular la cantidad media de huevos que pone cada gallina. Una muestra de 20 gallinas indica que ponen un promedio de 20 huevos al mes, con una desviación estándar de 2 huevos al mes.

In [8]:
muestra_gallinas=20
promedio_huevos=20
desviacion=2

### a) ¿Cuál es el valor de la media de la población? ¿Cuál es el mejor estimador de este valor? 

El problema no tiene media de población, para este ejemplo usaremos una media muestral que en este caso será 20 huevos promedio

### b) Explique por qué necesita utilizar la distribución t. ¿Qué suposiciones necesita hacer?

Usamos distribución t porque nuestra muestra es pequeña (menor a 30) y porque no tenemos desviacion poblacional, solo desviacion muestral

### c) ¿Cuál es el valor de t en un intervalo de confianza de 95%?

In [9]:
grado_de_libertad=muestra_gallinas-1
confianza=0.95
alfa=0.05
valor_t=st.t.ppf(1-alfa/2,grado_de_libertad)
print(f'El valor t para el ejmplo dado es {round(valor_t,3)}')

El valor t para el ejmplo dado es 2.093


### d) Construya el intervalo de confianza de 95% de la media de población.

In [10]:
limite_superior=promedio_huevos + (valor_t*(desviacion/math.sqrt(muestra_gallinas)))
limite_inferior=promedio_huevos - (valor_t*(desviacion/math.sqrt(muestra_gallinas)))
print(f'Los intérvalos de cofianza al {confianza*100}% de la media de población son: Lim Inferior {round(limite_inferior,2)} y Lim Superior {round(limite_superior,2)}')

Los intérvalos de cofianza al 95.0% de la media de población son: Lim Inferior 19.06 y Lim Superior 20.94


### e) ¿Es razonable concluir que la media poblacional es de 21 huevos? ¿Y de 25 huevos?

No es razonable que la media poblacional sea **ni 21 ni 25 huevos**, ya que vimos que los intervalos son máximo hasta 20.94
-   Talvez podriamos redondear los 20.94 a 21 huevos, pero sería ser generoso con la respuesta.
-   Para el promedio de 25 ya esta muy lejos y no podemos considerarlo

## 2. Sean  las  siguientes  hipótesis: 𝐻0:𝜇 =100; 𝐻1: 𝜇 ≠100 ,  una  muestra aleatoria de seis elementos dio como resultado los siguientes valores: 118, 105, 112, 119, 105 y 111. ¿Puede concluir que la media poblacional es diferente de 100 con un nivel de significancia de 0.05?

In [11]:
def desviacion_estandar_muestral(datos:np.array)->float:
    n=len(datos)
    u = sum(datos) / n

    return math.sqrt(sum((x - u)**2 for x in datos) / (n - 1))

In [12]:
valores=np.array([118, 105, 112, 119, 105 , 111])
muestra=len(valores)
alpha=0.05
confianza=0.95

media_muestral=valores.sum()/muestra
desviacion_muestral=desviacion_estandar_muestral(valores)

muestra,alpha,confianza,media_muestral,desviacion_muestral

(6, 0.05, 0.95, np.float64(111.66666666666667), 6.0553007081949835)

### a) Establezca la regla de decisión.

In [13]:
grados_de_libertad=muestra-1
valor_critico_sup = st.t.ppf(1 - alpha/2, grados_de_libertad)
valor_critico_inf = st.t.ppf( alpha/2, grados_de_libertad)
print(f'Definimos que los valores críticos de decisión son {round(valor_critico_inf,4)} inferior y {round(valor_critico_sup,4)} superior')


Definimos que los valores críticos de decisión son -2.5706 inferior y 2.5706 superior


### b) Calcule el valor del estadístico de prueba.

In [14]:
mu0 = 100

error_estandar = desviacion_muestral / math.sqrt(muestra)
t_estadistico = (media_muestral - mu0) / error_estandar
print(f'El valor estadístico de prueba calculado es : {round(t_estadistico,4)}')

El valor estadístico de prueba calculado es : 4.7194


### c) ¿Cuál es su decisión en lo que se refiere a la hipótesis nula?

In [15]:
if (t_estadistico > valor_critico_sup) or (t_estadistico < valor_critico_inf):
    print("Decisión: Rechazamos la hipótesis nula H0.")
else:
    print("Decisión: No rechazamos la hipótesis nula H0.")


Decisión: Rechazamos la hipótesis nula H0.


## 3. La familia Damon posee un viñedo grande en el oeste de Nueva York a orillas de lagoErie. Los viñedos deben fumigarse al inicio de la temporada de cultivo para protegerlos contra diversos insectos y enfermedades. Dos nuevos insecticidas acaban de salir al mercado: Pernod 5 y Action. Para probar su eficacia, se seleccionaron tres hileras y se fumigaron  con  Pernod  5,  y  otras  tres  se  fumigaron  con  Action.  Cuando  las  uvas maduraron, se revisaron 400 vides tratadas con Pernod 5 para saber si no estaban infectadas. De igual forma, se revisó una muestra de 400 vides fumigadas con Action. Los resultados son:

In [16]:
muestra_Pernod5=400
muestra_Action=400
vides_infectadas_Pernod5=24
vides_infectadas_Action=40

insignificancia=0.05

p1 = vides_infectadas_Pernod5 / muestra_Pernod5
p2 = vides_infectadas_Action / muestra_Action
print("p1 (Pernod 5):", p1)
print("p2 (Action):", p2)

p1 (Pernod 5): 0.06
p2 (Action): 0.1


Con un nivel de significancia de 0.05, ¿se puede concluir que existe una diferencia entre 
la  proporción  de  vides  infectadas  empleando  Pernod  5  en  comparación  con  las 
fumigadas con Action?

Vamos a formular una hipótesis nula

$$H_0: p_1 = p_2$$

y una hipótesis alternativa

$$H_1: p_1 \neq p_2$$

### a) formule la regla de decisión

In [17]:
z_crit = st.norm.ppf(1 - insignificancia/2)
print("Valor crítico z:", z_crit)


Valor crítico z: 1.959963984540054


### b) calcule la estimación conjunta de la varianza de la población

In [19]:
p_pool = (vides_infectadas_Pernod5 + vides_infectadas_Action) / (muestra_Pernod5 + muestra_Action)
var_pool = p_pool * (1 - p_pool) * (1/muestra_Pernod5 + 1/muestra_Action)
SE = math.sqrt(var_pool)

print("Proporción conjunta:", p_pool)
print("Error estándar conjunto:", SE)


Proporción conjunta: 0.08
Error estándar conjunto: 0.01918332609325088


### c) calcule el estadístico de prueba

In [20]:
z_stat = (p1 - p2) / SE
print("Estadístico z:", z_stat)


Estadístico z: -2.085144140570748


### d) tome una decisión respecto de la hipótesis nula 

In [21]:
if abs(z_stat) > z_crit:
    decision = "Rechazamos H0: Existe diferencia significativa entre las proporciones."
else:
    decision = "No se rechaza H0: No hay evidencia suficiente de diferencia en las proporciones."
    
print("Decisión:", decision)

Decisión: Rechazamos H0: Existe diferencia significativa entre las proporciones.
